diff --git a/nacl-configure b/nacl-configure
index 0472adc..4e10c9a 100755
--- a/nacl-configure
+++ b/nacl-configure
@@ -194,7 +194,7 @@ if [[ $TOOLCHAIN != "host" ]]; then
     NACL_CONFIGURE_LINE="$NACL_CONFIGURE_LINE -no-sse2"
 fi
 # misc
-NACL_CONFIGURE_LINE="$NACL_CONFIGURE_LINE -no-qpa-platform-guard -skip xmlpatterns -no-warnings-are-errors -no-qml-debug -no-dbus"
+NACL_CONFIGURE_LINE="$NACL_CONFIGURE_LINE -no-qpa-platform-guard -no-warnings-are-errors -no-qml-debug -no-dbus"
 
 # Newlib is static builds only.
 if [[ $QT_MKSPEC == *newlib* ]]

diff --git a/src/corelib/kernel/qfunctions_nacl.cpp b/src/corelib/kernel/qfunctions_nacl.cpp
index a5d86c9..a143d80 100644
--- a/src/corelib/kernel/qfunctions_nacl.cpp
+++ b/src/corelib/kernel/qfunctions_nacl.cpp
@@ -99,12 +99,12 @@ int pthread_attr_getinheritsched(const pthread_attr_t *, int *)
     return 0;
 }
 
-int pthread_condattr_init(pthread_condattr_t *)
+int pthread_condattr_init(pthread_condattr_t *) __attribute__((weak))
 {
     return 0;
 }
 
-int pthread_condattr_destroy(pthread_condattr_t *attr)
+int pthread_condattr_destroy(pthread_condattr_t *attr) __attribute__((weak))
 {
     return 0;
 }
@@ -308,12 +308,12 @@ int pthread_attr_setstacksize(pthread_attr_t *attr, size_t stacksize)
     return 0;
 }
 
-int pthread_condattr_init(pthread_condattr_t *)
+int pthread_condattr_init(pthread_condattr_t *) __attribute__((weak))
 {
     return 0;
 }
 
-int pthread_condattr_destroy(pthread_condattr_t *attr)
+int pthread_condattr_destroy(pthread_condattr_t *attr) __attribute__((weak))
 {
     return 0;
 }
